java-checks-test-sources/src/main/java/checks/spring/ValueAnnotationShouldInjectPropertyOrSpELCheckSample.java

```diff
@@ -0,0 +1,48 @@
+package checks.spring;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+class ValueAnnotationShouldInjectPropertyOrSpELCheckSample {
+
+  @Value("catalog.name") // Noncompliant [[sc=3;ec=25]] {{Either replace the "@Value" annotation with a standard field initialization, use "${propertyname}" to inject a property or use "#{expression}" to evaluate a SpEL expression.}}
+  String catalogA;
+
+  @Value("${catalog.name}") // Compliant
+  String catalogB;
+
+  @Value("book.topics[0]") // Noncompliant, this will not evaluate the expression
+  String topicA;
+
+  @Value("#{book.topics[0]}") // Compliant
+  String topicB;
+
+  @Value("Hello, world!") // Noncompliant, this use of @Value is redundant
+  String greetingA;
+
+  String greetingB = "Hello, world!"; // Compliant
+
+  @Value("") // Noncompliant
+  String empty;
+
+  public void setValue(@Value("xxx") String x){ // compliant
```

Comment:
Are we missing a noncompliant case where the annotation is applied to a parameter?